Brokerages/InteractiveBrokers/InteractiveBrokersBrokerage.cs

```diff
@@ -2481,62 +2481,83 @@ private DateTime GetRealTimeTickTime(Symbol symbol)
 
         private void HandleTickPrice(object sender, IB.TickPriceEventArgs e)
         {
+            // tickPrice events are always followed by tickSize events,
+            // so we save off the bid/ask/last prices and only emit ticks in the tickSize event handler.
+
             Symbol symbol;
 
-            if (!_subscribedTickets.TryGetValue(e.TickerId, out symbol)) return;
+            if (!_subscribedTickets.TryGetValue(e.TickerId, out symbol))
+            {
+                return;
+            }
 
             var price = Convert.ToDecimal(e.Price);
 
-            var tick = new Tick();
-            // in the event of a symbol change this will break since we'll be assigning the
-            // new symbol to the permtick which won't be known by the algorithm
-            tick.Symbol = symbol;
-            tick.Time = GetRealTimeTickTime(symbol);
-            var securityType = symbol.ID.SecurityType;
-            tick.Value = price;
-
-            if (e.Price <= 0 &&
-                securityType != SecurityType.Future &&
-                securityType != SecurityType.Option)
-                return;
-
-            int quantity;
+            Tick tick;
             switch (e.Field)
             {
                 case IBApi.TickType.BID:
                 case IBApi.TickType.DELAYED_BID:
 
-                    tick.TickType = TickType.Quote;
-                    tick.BidPrice = price;
-                    _lastBidSizes.TryGetValue(symbol, out quantity);
-                    tick.Quantity = quantity;
-                    _lastBidPrices[symbol] = price;
+                    if (!_lastQuoteTicks.TryGetValue(symbol, out tick))
+                    {
+                        tick = new Tick
+                        {
+                            // in the event of a symbol change this will break since we'll be assigning the
+                            // new symbol to the permtick which won't be known by the algorithm
+                            Symbol = symbol,
+                            TickType = TickType.Quote
+                        };
+                        _lastQuoteTicks.TryAdd(symbol, tick);
+                    }
+
+                    // negative price (-1) means no price available, normalize to zero
+                    tick.BidPrice = price < 0 ? 0 : price;
```

Comment:
Maybe we could normalize the price outside the switch so it's shared? (doesn't it also apply for `IBApi.TickType.LAST`?

Reply:
I've never seen a negative price for `IBApi.TickType.LAST` but will move the normalization up to remove duplication